#!/bin/bash

# Runs per sample portion of bismark methylation analysis of custom amplicon experiment
# Author: James Boot, Date: 16/02/2023
# Based on original scripts by Ian Donaldson and Anna Terry

# In the future run this in the dedup_results folder 

# SECTION 1: EDIT HERE

##### Set queue options
#$ -cwd
#$ -j y
#$ -pe smp 4
#$ -l h_rt=240:0:0
#$ -l h_vmem=8G
#$ -N dedup-test_GC-AS-10657					# Add traceable name
#$ -t 1-48					# Edit for the number of samples running (1-N)
#####

# SECTION 2: 
# INPUT_DIR should be the bismark output folder
# OUTPUT_DIR should be the decired dedup results folder
# Check how samples have been named and saved, two sample name parameters files may be needed if the folder name and the file name do not match - sometimes illumina S[1-9] is included in the file name but not folder name
# INFILE should point to the bam file output from bismark - check naming convention
# OUTFILE should be the desired file name of the intermediate file generated by samtools - note this is not the final output!
source /data/WHRI-GenomeCentre/shares/Projects/NGS_Projects/DNA_Sequencing/Sharples_Adam/GC-AS-10657/Analysis/4.collectFiles/config.txt
INPUT_DIR=/data/WHRI-GenomeCentre/shares/Projects/NGS_Projects/DNA_Sequencing/Sharples_Adam/GC-AS-10657/Analysis/6.bismark_test_results
OUTPUT_DIR=/data/WHRI-GenomeCentre/shares/Projects/NGS_Projects/DNA_Sequencing/Sharples_Adam/GC-AS-10657/Analysis/7.dedup_test_results
SAMPLE1=$(sed -n -e "$SGE_TASK_ID p" /data/WHRI-GenomeCentre/shares/Projects/NGS_Projects/DNA_Sequencing/Sharples_Adam/GC-AS-10657/Analysis/sample_name_list.txt)
SAMPLE2=$(sed -n -e "$SGE_TASK_ID p" /data/WHRI-GenomeCentre/shares/Projects/NGS_Projects/DNA_Sequencing/Sharples_Adam/GC-AS-10657/Analysis/2.MergeRuns/sample_names.txt)
INFILE=${INPUT_DIR}/${SAMPLE1}/${SAMPLE2}_R1_UMI_val_1_bismark_bt2_pe.bam
OUTFILE1=${INPUT_DIR}/${SAMPLE1}/${SAMPLE2}_R1_processed_val_1_bismark_bt2_pe_dedup.bam

# Check output folder exists
if [ ! -e ${OUTPUT_DIR} ]; then mkdir -p ${OUTPUT_DIR}; fi

# Load modules
module load gcc/7.1.0
module load bowtie2/2.4.1
module load samtools/1.6
module load bismark/0.22.1

# Remove the index sequences and extra bits from the end of the read name
samtools view -h ${INFILE} | sed 's/_[1-2]:[A-Z]:[0-9]:[A-Z][A-Z][A-Z][A-Z][A-Z][A-Z][A-Z][A-Z][+][A-Z][A-Z][A-Z][A-Z][A-Z][A-Z][A-Z][A-Z]//g' | samtools view -o ${OUTFILE1}

# Run Bismark dedup
# Final output files are output appended with .deduplicated.bam and .deduplication_report.txt and saved in working directory - move to appropriate folder on completion
deduplicate_bismark -p --barcode ${OUTFILE1}
